src/x/checked/bytes_test.go

```diff
@@ -60,6 +62,20 @@ func TestBytes(t *testing.T) {
 
 	finalizerCalls := 0
 	onFinalize = func(finalizing Bytes) {
+		// Ensure that RefCount.Finalize was called before the bytesRef finalizer.
+		stack := debug.Stack()
+		// TODO: check if RefCount.Finalize was called before the bytesRef finalizer
+		// instead of just printing stack.
+		// i.e. this looks good:
+		/*
+			github.com/m3db/m3/src/x/checked.(*bytesRef).OnFinalize(0xc00009e080)
+				/Users/r/go/src/github.com/m3db/m3/src/x/checked/bytes.go:129 +0x81
+			github.com/m3db/m3/src/x/checked.(*RefCount).Finalize(0xc00009e080)
+				/Users/r/go/src/github.com/m3db/m3/src/x/checked/ref.go:77 +0xbd
+			github.com/m3db/m3/src/x/checked.TestBytes(0xc00011e100)
+		*/
+		fmt.Printf("calling stack: %s\n", stack)
```

Comment:
Could you add a check here that this order is respected?

This isn't quite good enough (since it doesn't check order, and only checks that the string is contained) but can we do something similar:
```
		assert.True(t, strings.Contains(str, "read after free: reads=1, ref=0"))
		assert.True(t, strings.Contains(str, "IncReads, ref=0, unixnanos="))
		assert.True(t, strings.Contains(str, "checked.(*RefCount).IncReads"))
		assert.True(t, strings.Contains(str, "DecRef, ref=0, unixnanos="))
		assert.True(t, strings.Contains(str, "checked.(*RefCount).DecRef"))
		assert.True(t, strings.Contains(str, "IncRef, ref=1, unixnanos="))
		assert.True(t, strings.Contains(str, "checked.(*RefCount).IncRef"))
```

For reference:
https://github.com/m3db/m3/blob/master/src/x/checked/debug_test.go/#L47-L102